packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -535,6 +538,43 @@ export function canvasReducer(
 
             break;
           }
+          case 'addSymbolLayer': {
+            const symbol = {
+              ...getSymbols(state).find(
+                ({ do_objectID }) => do_objectID === interactionState.symbolId,
+              ),
+            } as Sketch.SymbolMaster;
+
+            const layer = produce(Models.symbolInstance, (layer) => {
+              if (!layer.style) return;
+              layer.do_objectID = interactionState.id;
+              layer.name = symbol.name;
+              layer.symbolID = symbol.symbolID;
+              layer.frame = {
+                ...symbol.frame,
+                x: interactionState.point.x - symbol.frame.width / 2,
+                y: interactionState.point.y - symbol.frame.height / 2,
+              };
+            });
+
+            // Check if the layer intersects any artboards or symbolMasters.
+            // If so, we'll insert the layer within
+            const parent = draft.sketch.pages[pageIndex].layers
+              .filter(
+                (layer): layer is Sketch.Artboard | Sketch.SymbolMaster =>
+                  Layers.isArtboard(layer) || Layers.isSymbolMaster(layer),
+              )
+              .find((artboard) => rectsIntersect(artboard.frame, layer.frame));
+
+            if (parent && Layers.isChildLayer(layer)) {
+              layer.frame.x -= parent.frame.x;
+              layer.frame.y -= parent.frame.y;
+
+              parent.layers.push(layer);
+            } else {
+              draft.sketch.pages[pageIndex].layers.push(layer);
+            }
+          }
```

Comment:
Let's also automatically select the newly created layer, `draft.selectedObjects = [layer.do_objectID]`